ultralytics/utils/__init__.py

```diff
@@ -262,12 +280,38 @@ def to_xml(self, normalize=False, decimals=5):
 
         Returns:
             (str): XML string.
-
-        Notes:
-            Requires `lxml` package to be installed.
         """
         df = self.to_df(normalize=normalize, decimals=decimals)
-        return '<?xml version="1.0" encoding="utf-8"?>\n<root></root>' if df.empty else df.to_xml(parser="etree")
+        if df.is_empty():
+            return '<?xml version="1.0" encoding="utf-8"?>\n<root></root>'
+
+        def _to_xml_str_simple(v):
+            """Convert basic types to XML-safe string."""
+            if v is None:
+                return ""
+            if isinstance(v, (dict, list, tuple, set)):
+                return str(v).replace("'", "&apos;").replace('"', "&quot;")
+            if isinstance(v, str):
+                return (
+                    v.replace("&", "&amp;")
+                    .replace("<", "&lt;")
+                    .replace(">", "&gt;")
+                    .replace("'", "&apos;")
+                    .replace('"', "&quot;")
+                )
+            return str(v)
+
+        xml_lines = ['<?xml version="1.0" encoding="utf-8"?>', "<root>"]
+
+        for row_dict in df.iter_rows(named=True):
+            xml_lines.append("  <row>")
+            for col_name, value in row_dict.items():
+                xml_value = _to_xml_str_simple(value)
+                xml_lines.append(f"    <{col_name}>{xml_value}</{col_name}>")
+            xml_lines.append("  </row>")
+
+        xml_lines.append("</root>")
+        return "\n".join(xml_lines)
 
     def to_html(self, normalize=False, decimals=5, index=False):
```

Comment:
Let's probably delete to_html() also. Same thing: never heard a user ask for HTML results.